shared/fs/common/load-path-metadata-when-needed.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import * as React from 'react'
+import {namedConnect} from '../../util/container'
+import * as FsGen from '../../actions/fs-gen'
+import * as Types from '../../constants/types/fs'
+
+type OwnProps = {|path: Types.Path|}
+
+const mapDispatchToProps = (dispatch, {path}) => ({
+  loadPathMetadata: () => dispatch(FsGen.createLoadPathMetadata({path})),
+})
+
+const mergeProps = (s, d, o) => ({
+  ...o,
+  ...d,
+})
+
+type Props = {|
+  loadPathMetadata: () => void,
+  path: Types.Path,
+|}
+
+class LoadPathMetadataWhenNeeded extends React.PureComponent<Props> {
+  componentDidMount() {
+    this.props.loadPathMetadata()
+  }
+  componentDidUpdate(prevProps) {
+    if (this.props.path !== prevProps.path) {
+      this.props.loadPathMetadata()
+    }
+  }
+  render() {
+    return null
```

Comment:
Wouldn't it make more sense for this to be a HOC?

Reply:
I think this is cleaner and easier to type. HOCs are necessary when we need to pass props to the wrapped component. In this case, it doesn't actually provide any props to a component; all we need is to dispatch actions at the right time. We do this with `fs/row/load-files-when-needed.js` too, and use that in `fs/row/rows.js`

Reply:
I prefer the HOC way since it better represents what is happening: it's triggering an event that should cause upstream props to be loaded. Whereas the current approach makes it look like it's a visible component that displays something.

But it's not a big deal.

Reply:
The way I look at it is it just triggers an action that causes side effect. This action or its side effect is not owned by any components. When we are ready to move to hooks, things like this could probably just use `useEffect`.